Tests/RoslynPluginGeneratorTests/AnalyzerPluginGeneratorTests.cs

```diff
@@ -207,20 +363,103 @@ public void Generate_LicenseAcceptanceNotRequestedIfNoAnalysers()
 
             RemoteRepoBuilder remoteRepoBuilder = new RemoteRepoBuilder(this.TestContext);
 
-            // Parent only: requires license
-            remoteRepoBuilder.CreatePackage("non.analyzer.requireAccept.id", "1.0", dummyContentFile, License.Required);
+            // Multi-level: parent and some but not all dependencies require license acceptance
+            // Parent and dependencies have no analyzers
+            IPackage child1 = remoteRepoBuilder.CreatePackage("child1.requiredAccept.id", "2.1", dummyContentFile, License.Required);
+            IPackage child2 = remoteRepoBuilder.CreatePackage("child2.id", "2.2", dummyContentFile, License.NotRequired);
+            remoteRepoBuilder.CreatePackage("non-analyzer.parent.requireAccept.id", "1.0", dummyContentFile, License.Required, child1, child2);
+
+            TestLogger logger = new TestLogger();
+            AnalyzerPluginGenerator apg = CreateTestSubjectWithFakeRemoteRepo(remoteRepoBuilder, logger);
+
+            // 1. a) Only target package. User does not accept, but no analyzers so no license prompt -> fails due to absence of analyzers
+            ProcessedArgs args = CreateArgs("non-analyzer.parent.requireAccept.id", "1.0", "cs", null, false /* accept licenses */ , 
+                false, outputDir);
+            bool result = apg.Generate(args);
+            Assert.IsFalse(result, "Expecting generator to fail");
+
+            logger.AssertSingleWarningExists(String.Format(UIResources.APG_NoAnalyzersFound, "non-analyzer.parent.requireAccept.id"));
+            logger.AssertWarningsLogged(1);
+            logger.AssertErrorsLogged(0);
+
+            // 1. b) Target package and dependencies. User does not accept, but no analyzers in any package so no license prompt -> fails due to absence of analyzers
+            logger.Reset();
+            args = CreateArgs("non-analyzer.parent.requireAccept.id", "1.0", "cs", null, false /* accept licenses */ ,
+                true /* generate plugins for dependencies */, outputDir);
+            result = apg.Generate(args);
+            Assert.IsFalse(result, "Expecting generator to fail");
+
+            logger.AssertSingleWarningExists(String.Format(UIResources.APG_NoAnalyzersFound, "non-analyzer.parent.requireAccept.id"));
+            logger.AssertSingleWarningExists(String.Format(UIResources.APG_NoAnalyzersFound, child1.Id));
+            logger.AssertSingleWarningExists(String.Format(UIResources.APG_NoAnalyzersFound, child2.Id));
+            logger.AssertWarningsLogged(3);
+            logger.AssertErrorsLogged(0);
+        }
+
+        [TestMethod]
+        public void Generate_LicenseAcceptanceRequested_NoAnalysersInTarget()
```

Comment:
This test does add value as there is an expected change in behaviour in this case.